src/network/network_client.cpp

```diff
@@ -1258,6 +1295,9 @@ void NetworkUpdateClientName()
 	NetworkClientInfo *ci = NetworkClientInfo::GetByClientID(_network_own_client_id);
 
 	if (ci == nullptr) return;
+	/* There is no validation on string settings. So, when the client name is changed via the console
+	 * there is no easy way to prevent an invalid name. Though, we can prevent it getting sent here. */
```

Comment:
Is this true? You can have validations and post-functions on settings that change. Can't we do these things there?

Reply:
https://github.com/OpenTTD/OpenTTD/blob/31c5b8fe0f9c070143625478f7120e0c9c2332fd/src/settings.cpp#L1327 for example, I think.

Reply:
This function is called in the post-function of the setting.
Yes, I can do this validation there. However, then I would be putting more networking logic in the post-function that is going to call this function, so putting it here feels better to me. Now it is also only going to complain when it is actually used.